src/Parser.js

```diff
@@ -1010,24 +1039,27 @@ export default class Parser {
      * @param {Array.<ParseNode>} group  the nodes of this group,
      *                                   list will be moified in place
      */
-    formLigatures(group) {
+    formLigatures(group: ParseNode[]) {
         let n = group.length - 1;
         for (let i = 0; i < n; ++i) {
             const a = group[i];
             const v = a.value;
             if (v === "-" && group[i + 1].value === "-") {
                 if (i + 1 < n && group[i + 2].value === "-") {
                     group.splice(i, 3, new ParseNode(
+                        // $FlowFixMe: update ParseNode to accept a location
                         "textord", "---", "text", a, group[i + 2]));
```

Comment:
The last two args are supposed to be `Token`s, but we're passing `ParseNode`s.  Unfortunately, there's no way to get the `Token`s here.

Reply:
Those parameter can be changed from `Token`s to `{loc: ?SourceLocation}`, just like the parameters of `SourceLocation.range`.